src/osd/PrimaryLogPG.cc

```diff
@@ -3991,15 +3991,16 @@ int PrimaryLogPG::do_extent_cmp(OpContext *ctx, OSDOp& osd_op)
     return result;
   }
 
-  if (read_op.outdata.length() != osd_op.indata.length())
-    return -EINVAL;
-
-  for (uint64_t p = 0; p < osd_op.indata.length(); p++) {
+  for (uint64_t p = 0; p < read_op.outdata.length(); p++) {
     if (read_op.outdata[p] != osd_op.indata[p]) {
       return (-MAX_ERRNO - p);
     }
   }
 
+  if (read_op.outdata.length() != osd_op.indata.length()) {
```

Comment:
@jdurgin @liewegas You guys have any thoughts about this? Other alternatives would be to keep the extent check in-place but (1) override it via a flag when issuing the ```cmpext``` op, or (2) perhaps allow ```cmpext``` to pass a smaller compare buffer than the write buffer (which will allow librbd to trim any trailing zeroes from the compare buffer before sending to the OSD).